services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/signing/DefaultS3Presigner.java

```diff
@@ -330,7 +344,15 @@ private <T extends PresignedRequest.Builder, U> T presign(T presignedRequest,
                                                               Function<U, SdkHttpFullRequest> requestMarshaller,
                                                               String operationName) {
 
-        ExecutionContext execCtx = invokeInterceptorsAndCreateExecutionContext(presignRequest, requestToPresign, operationName);
+        // A fixed signingClock is used, so that the current time used by the signing logic, as well as to determine expiration
+        // are the same.
+        Instant signingInstant = Instant.now();
+        Clock signingClock = Clock.fixed(signingInstant, ZoneOffset.UTC);
```

Comment:
An alternative to this would be to let the HttpSigner use current time inside it's logic, and later to calculate `expiration` to set inside presignedRequest in `initializePresignedRequest` we could extract the current time value used by the HttpSigner by reading the value out of `X-Amz-Date` query param inserted by the HttpSigner. `initializePresignedRequest` already has some logic to extract from `X-Amz-SignedHeaders` query param. But it seemed better to not rely more on implementation details of the HttpSigner.